examples/compute_polar_coordinates.py

```diff
@@ -0,0 +1,362 @@
+    angles="xy",
+    scale=1,
+    scale_units="xy",
+    headwidth=7,
+    headlength=9,
+    headaxislength=9,
+    color="gray",
+)
+
+ax.axis("equal")
+ax.set_xlim(xmin, xmin + x_delta)
+ax.set_ylim(ymin, ymin + y_delta)
+ax.set_xlabel("x (pixels)")
+ax.set_ylabel("y (pixels)")
+ax.set_title(f"Zoomed in to check ({mouse_name})")
+ax.invert_yaxis()
+fig.colorbar(sc, ax=ax, label=f"time ({ds.attrs['time_unit']})")
+
+ax.legend(
+    [
+        "midpoint_ears",
+        "snout",
+        "head_vector",
+    ],
+    loc="best",
+)
+
+fig.show()
+
+# %%
+# From the plot we can confirm the head vector goes from the midpoint between
+# the ears to the snout, as we defined it.
+
+
+# %%
+# Express the head vector in polar coordinates
+# -------------------------------------------------------------
+# A convenient way to inspect the orientation of a vector in 2D is by
+# expressing it in polar coordinates. We can do this with the vector function
+# ``cart2pol``:
+head_vector_polar = cart2pol(head_vector)
+
+print(head_vector_polar)
+
+# %%
+# Notice how the resulting array has a ``space_pol`` dimension with two
+# coordinates: ``rho`` and ``phi``. These are the polar coordinates of the
+# head vector.
+#
+# The coordinate ``rho`` is the norm (length) of the vector. In our case, the
+# distance from the midpoint between the ears to the snout. The coordinate
+# ``phi`` is the orientation of the head vector relative to the positive
+# x-axis, and ranges from -``pi`` to ``pi``
+# (following the [atan2](https://en.wikipedia.org/wiki/Atan2) convention).
+#
+# In our coordinate system, ``phi`` will be
+# positive if the shortest path from the positive x-axis to the vector is
+# clockwise. Conversely, ``phi`` will be negative if the shortest path from
+# the positive x-axis to the vector is anti-clockwise.
+
+# %%
+# Histogram of ``rho`` values
+# ----------------------------
+# Since ``rho`` is the distance between the ears' midpoint and the snout,
+# we would expect ``rho`` to be approximately constant in this data. We can
+# check this by plotting a histogram of its values across the whole clip.
+
+fig, ax = plt.subplots(1, 1)
+
+rho_data = head_vector_polar.sel(individuals=mouse_name, space_pol="rho")
+
+# compute and plot histogram
+ax.hist(
+    rho_data,
+    bins=50,
+)
+
+# add mean
+ax.axvline(
+    x=np.nanmean(rho_data),
+    c="b",
+    linestyle="--",
+)
+
+
+# add median
+ax.axvline(
+    x=np.nanmedian(rho_data),
+    c="r",
+    linestyle="-",
+)
+
+# add legend
+ax.legend(
+    [
+        f"mean = {np.nanmean(rho_data):.2f} pixels",
+        f"median = {np.nanmedian(rho_data):.2f} pixels",
+    ],
+    loc="best",
+)
+ax.set_ylabel("count")
+ax.set_xlabel(r"$\rho$" + " (pixels)")
+fig.show()
+
+# %%
+# We can see that there is some spread in the value of ``rho`` in this
+# dataset. This may be due to noise in the detection of the head keypoints,
+# or due to the mouse tipping its snout upwards during the recording.
```

Comment:
or downwards, right? (which is the more common occurrence in this particular video)

Reply:
but downwards wouldn't detect the snout right?

Reply:
yeap, in that case it would disappear.